### PR TITLE
Fix integration tests on Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?

Fixes integration tests on Windows... to include the horrible memory leak.
### What issues does this PR fix or reference?

Part of this: https://github.com/saltstack/zh/issues/653
### Previous Behavior

Tests would run but fail to communicate with the minion, so it wasn't really working. Memory would be consumed until all was consumed.
### New Behavior

Tests actually run. No memory leak.
### Tests written?

They're about to be...

### CODE REVIEW

**Comment by cachedout** [comment]
This breaks the test suite on Linux.

**Comment by twangboy** [comment]
Rebased....

File: tests/integration/__init__.py [hunk 1]
```diff
@@ -979,30 +980,30 @@ def transplant_configs(cls, transport='zeromq'):
             running_tests_user = win32api.GetUserName()
         else:
             running_tests_user = pwd.getpwuid(os.getuid()).pw_name
-        master_opts = salt.config._read_conf_file(os.path.join(CONF_DIR, 'master'))
```
Comment (s0undt3ch) [inline tests/integration/__init__.py]: If you can't have the user setting on windows, make this setting check for that, but don't remove it.

Comment (s0undt3ch) [inline tests/integration/__init__.py]: Heh nevermind, you just placed it further down....

File: tests/integration/files/conf/sub_minion [hunk 2]
```diff
@@ -33,6 +34,3 @@ grains:
       - jamie
       - zoe
 
-ipc_mode: tcp
```
Comment (s0undt3ch) [inline tests/integration/files/conf/sub_minion]: Leave the ipc_mode setting as it was, sub minion was purposely using TCP, even on Linux.

File: tests/integration/files/conf/sub_minion [hunk 3]
```diff
@@ -1,15 +1,16 @@
 # basic config
 master: localhost
 master_port: 64506
+tcp_pub_port: 64510
+tcp_pull_port: 64511
 root_dir: /tmp/subsalttest
-pki_dir: pki
-id: sub_minion
-cachedir: cachedir
```
Comment (s0undt3ch) [inline tests/integration/files/conf/sub_minion]: Was the cachedir setting also part of the problem?

File: tests/integration/__init__.py [hunk 4]
```diff
@@ -725,6 +725,7 @@ def start_zeromq_daemons(self):
         self.syndic_process = SaltSyndic(self.syndic_opts, TMP_SYNDIC_MINION_CONF_DIR, SCRIPT_DIR)
         self.syndic_process.display_name = 'salt-syndic'
         for process in (self.master_process, self.minion_process, self.sub_minion_process,
+        #for process in (self.master_process, self.minion_process,
```
Comment (s0undt3ch) [inline tests/integration/__init__.py]: This commented line could be removed

File: tests/integration/files/conf/master [hunk 5]
```diff
@@ -5,19 +5,16 @@ publish_port: 64505
 ret_port: 64506
 worker_threads: 3
 root_dir: /tmp/salttest
-pidfile: masterpid
-pki_dir: pki
```
Comment (s0undt3ch) [inline tests/integration/files/conf/master]: Don't remove the pki_dir  setting.

**Comment by s0undt3ch** [comment]
Basically, leave the configuration files the same they were and add the TCP ports and interface for the minion.

The pidfile changes and the log file changes are OK.

I believe that with only these changes, we'll also have a non leaking test suite on windows...

**Comment by cachedout** [comment]
This looks like a bad merge. Please rebase this or close and re-open with only your intended commits.

File: tests/integration/files/conf/master [hunk 6]
```diff
@@ -1,24 +1,25 @@
 id: master
-user: ubuntu
 interface: 127.0.0.1
 publish_port: 64505
 ret_port: 64506
 worker_threads: 3
 pidfile: master.pid
+user: ubuntu
```
Comment (s0undt3ch) [inline tests/integration/files/conf/master]: The user should not be set, the test suite takes care of identifying the user running the tests and adding that to the configuration which is generated.

**Comment by s0undt3ch** [comment]
I think the PKI setting that was removed should probably remain as it was, unless, that's what's breaking the test runs...

**Comment by s0undt3ch** [comment]
The same for cachedir

**Comment by s0undt3ch** [comment]
So, we detected 2 problems here:
1. Salt shall not leak when unable to connect, this is a serious bug;
2. We need to specify the pki_dir because the default is an absolute path, and since its absolute, it won't get prefixed with root_dir. Leaving the default value might make the test suite return inconsistent results.

**Comment by twangboy** [comment]
Rebased

**Pull request closed without merging**